src/main/webapp/authorization.js

```diff
@@ -1,56 +1,124 @@
+   * purposes, this function will serve as a true/false value in an if
+   * statement.
+   *
+   * @return {Firebase.User} current user or null
+   */
+  getUser() {
+    return this.firebase.auth().currentUser;
+  }
 
-/**
- * Function called when clicking the Login/Logout button.
- */
-function toggleSignIn() {
-  const firebase = getFirebase();
-  if (!firebase.auth().currentUser) {
-    let provider = new firebase.auth.GithubAuthProvider();
-    firebase.auth().signInWithPopup(provider).then(function(result) {
-      // This gives you a GitHub Access Token.
-      // You can use it to access the GitHub API.
-      const token = result.credential.accessToken; //eslint-disable-line
-      // The signed-in user info.
-      const user = result.user; //eslint-disable-line
-    }).catch(function(error) {
-      // Handle Errors here.
-      console.error(error);
-      // Change this in future iterations
-      alert(error.message);
-    });
-  } else {
-    // if logged in, sign out
-    firebase.auth().signOut();
+  /**
+   * Get firebase instance used with this object
+   * @return {Firebase} firebase instance
+   */
+  getFirebase() {
+    return this.firebase;
+  }
+
+  /**
+   * Function called when authorizing user to verify them as a gitHub user.
+   * This function doubles as an opt out when the current user wants to leave
+   * our site for good.
+   *
+   * Side Effects: when authorizing a user, it sets the current objects token
+   * to the returned GitHub API accessToken if a token is returned.
+   *
+   * NOTE: this function is asynchronous and should be used with an await
+   *
+   * @return {boolean} true if no errors occured during authorization, false
+   * otherwise
+   */
+  async toggleAuthorization() {
+    const firebase = this.firebase;
+    if (!firebase.auth().currentUser) {
+      try {
+        let provider = new firebase.auth.GitHubAuthProvider();
+        let authorizationResults =
+          await firebase.auth().signInWithPopup(provider);
+        // This gives you a GitHub Access Token.
+        // You can use it to access the GitHub API.
+        this.token = await authorizationResults.credential.accessToken;
+      } catch (error) {
+        // Handle Errors here.
+        console.error(error);
+        // TODO: Change this in future iterations
+        alert(error.message);
+        this.token = null;
+        return false
+      }
+    } else {
+      // if logged in, sign out
+      firebase.auth().signOut();
+      this.token = null;
+    }
+    return true
+  }
+
+  /**
+   * Initialize firebase if firebase is not already initialized. This method can
+   * be called as many times as we need without any side effects.
+   *
+   * NOTE: This is not an instance function and will not be exported.
+   */
+  static initializeFirebase() {
+    if (firebase.apps.length === 0) { // eslint-disable-line
+      firebase.initializeApp(firebaseConfig); // eslint-disable-line
```

Comment:
```suggestion
 initializeFirebase() {
    if (this.getFirebase().apps.length === 0) {
      this.getFirebase().initializeApp(firebaseConfig);
```
Does this have to be static? Or does using getFirebase() work here